libs/staging-provider/src/Provider/Configuration/WorkspaceBackendConfig.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\StagingProvider\Provider\Configuration;
+
+use Keboola\InputMapping\Staging\AbstractStrategyFactory;
+use Keboola\StagingProvider\Exception\StagingProviderException;
+
+readonly class WorkspaceBackendConfig
+{
+    public function __construct(
+        private string $stagingType,
+        private ?string $size,
+        private ?bool $useReadonlyRole,
+    ) {
+    }
+
+    public function getStagingType(): string
+    {
+        return $this->stagingType;
+    }
+
+    public function getStorageApiWorkspaceType(): string
+    {
+        return match ($this->stagingType) {
+            AbstractStrategyFactory::WORKSPACE_ABS => 'abs',
+            AbstractStrategyFactory::WORKSPACE_BIGQUERY => 'bigquery',
+            AbstractStrategyFactory::WORKSPACE_EXASOL => 'exasol',
+            AbstractStrategyFactory::WORKSPACE_REDSHIFT => 'redshift',
+            AbstractStrategyFactory::WORKSPACE_SNOWFLAKE => 'snowflake',
+            AbstractStrategyFactory::WORKSPACE_SYNAPSE => 'synapse',
+            AbstractStrategyFactory::WORKSPACE_TERADATA => 'teradata',
+            default => throw new StagingProviderException(sprintf('Unknown staging type "%s"', $this->stagingType)),
+        };
```

Comment:
By se dalo predelat ty konstanty do enumu a pak se toho tu zbavit.

Reply:
Nedalo, to je z IM a používá se to všude možně :)

Reply:
Jo, to jsem nemyslel ze se ma udelat, jen ze kdyby se to nekdy udelalo (vim ze je to v IM)